src/pages/Admin/Proposal/Voter/useVote.ts

```diff
@@ -1,40 +1,45 @@
-import { MsgExecuteContractEncodeObject } from "@cosmjs/cosmwasm-stargate";
 import { useFormContext } from "react-hook-form";
 import { VoteValues as VV } from "./types";
-import { useAdminResources } from "pages/Admin/Guard";
+import { SimulContractTx } from "types/evm";
 import { invalidateJunoTags } from "services/juno";
 import { adminTags } from "services/juno/tags";
+import { useErrorContext } from "contexts/ErrorContext";
 import { useGetWallet } from "contexts/WalletContext";
-import CW3 from "contracts/CW3";
-import CW3Review from "contracts/CW3/CW3Review";
+import { MultiSigGeneric } from "contracts/evm";
 import useTxSender from "hooks/useTxSender";
+import { WalletDisconnectedError } from "errors/errors";
+import { polygonContracts } from "constants/contracts";
 
 export default function useVote() {
   const {
     handleSubmit,
     formState: { isValid },
   } = useFormContext<VV>();
   const { wallet } = useGetWallet();
-  const { cw3 } = useAdminResources();
   const { sendTx, isSending } = useTxSender(true);
+  const { handleError } = useErrorContext();
 
-  async function vote({ type, proposalId, vote, reason }: VV) {
-    let voteMsg: MsgExecuteContractEncodeObject;
-    if (type === "application") {
-      const contract = new CW3Review(wallet);
-      voteMsg = contract.createVoteApplicationMsg({
-        proposal_id: proposalId,
-        vote: vote,
-        reason: (vote === "no" && reason) || undefined,
-      });
-    } else {
-      //normal cw3
-      const contract = new CW3(wallet, cw3);
-      voteMsg = contract.createVoteMsg(proposalId, vote);
+  async function vote({ type, proposalId: transactionId, vote }: VV) {
+    if (!wallet) {
+      return handleError(new WalletDisconnectedError());
     }
 
+    const toAddress =
+      type === "application"
+        ? polygonContracts.multiSig.ApplicationsMultiSigProxy
+        : polygonContracts.multiSig.APTeamMultiSigProxy;
+
+    const voteMsg: SimulContractTx = {
+      from: wallet.address,
+      to: toAddress,
+      data:
+        vote === "no"
+          ? MultiSigGeneric.revokeTransaction.encode(transactionId)
```

Comment:
@misicnenad, It's not part of current UX to allow users to revoke already confirmed transactions, or should we allow them to @SovereignAndrey  ?

Reply:
@SovereignAndrey @ap-justin this is actually a good question. Since our UI allows for voting "no" on transactions (in Juno called "proposals"), but in our contracts a "no" vote **does not exist** (abstaining is considered a "no" vote), we could make it so that:
1. initially only voting "yes" or abstaining is possible.
2. if the user already voted on a transaction (with a "yes" vote) and the transaction can still be voted on, they are able to change their vote to "no" in order to revoke their vote.

Reply:
@SovereignAndrey on the other hand, even if wanted to do this, it should be treated as a separate task/PR, as it includes changing the actual UI/UX logic. 
Will remove the ability to vote "no" and revoke transactions for now, but we can look into updating the flow as described in [my previous comment](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2023/files#r1160564271).

Reply:
Terminology of "yes" and "no" makes sense in the context of voting on proposals. On the other hand, in the context of "submitting a signature" to pass a multisig transaction, a terminology that would make a little more sense would probably be "submit signature" and "revoke signature".

Reply:
@misicnenad & @ap-justin: 

> @misicnenad, It's not part of current UX to allow users to revoke already confirmed transactions, or should we allow them to @SovereignAndrey ?

Yes, we have the ability to revoke a previously confirmation, so we should think about how that would look. The Decision Center UI _might_ have something to point us in the right direction. 

> @SovereignAndrey on the other hand, even if wanted to do this, it should be treated as a separate task/PR,

Yes it should be covered in a separate PR since it's an bigger enhancement with wider UI changes that will need to come with it. 

> Terminology of "yes" and "no" makes sense in the context of voting on proposals. On the other hand, in the context of "submitting a signature" to pass a multisig transaction, a terminology that would make a little more sense would probably be "submit signature" and "revoke signature".

"Confirm" and "Revoke Confirmation" makes more sense than "yes"/"no". We'd also only want to show "Confirm" option first and then only show "Revoke Confirmation" once a confirmation was signed to a TX. Having both options shown at once does not make sense.